extension/chrome/settings/modules/security.ts

```diff
@@ -7,122 +7,135 @@ import { Store } from '../../../js/common/platform/store.js';
+    this.acctEmail = Assert.urlParamRequire.string(uncheckedUrlParams, 'acctEmail');
+    this.parentTabId = Assert.urlParamRequire.string(uncheckedUrlParams, 'parentTabId');
+    this.embedded = uncheckedUrlParams.embedded === true;
+  }
 
-Catch.try(async () => {
-
-  const uncheckedUrlParams = Url.parse(['acctEmail', 'embedded', 'parentTabId']);
-  const acctEmail = Assert.urlParamRequire.string(uncheckedUrlParams, 'acctEmail');
-  const parentTabId = Assert.urlParamRequire.string(uncheckedUrlParams, 'parentTabId');
-  const embedded = uncheckedUrlParams.embedded === true;
-
-  await initPassphraseToggle(['passphrase_entry']);
+  async render() {
+    await initPassphraseToggle(['passphrase_entry']);
+    [this.primaryKi] = await Store.keysGet(this.acctEmail, ['primary']);
+    Assert.abortAndRenderErrorIfKeyinfoEmpty(this.primaryKi, false);
+    if (!this.primaryKi) {
+      return; // added do_throw=false above + manually exiting here because security.htm can indeed be commonly rendered on setup page before setting acct up
+    }
+    if (this.embedded) {
+      $('.change_passhrase_container, .title_container').css('display', 'none');
+      $('.line').css('padding', '7px 0');
+    }
+    this.authInfo = await Store.authInfo(this.acctEmail);
+    const storage = await Store.getAcct(this.acctEmail, ['hide_message_password', 'outgoing_language']);
+    $('#hide_message_password').prop('checked', storage.hide_message_password === true);
+    $('.password_message_language').val(storage.outgoing_language || 'EN');
+    await this.renderPPOptionsIfStoredInSession();
+    await this.renderSubscriptionDetails();
+  }
 
-  const [primaryKi] = await Store.keysGet(acctEmail, ['primary']);
-  Assert.abortAndRenderErrorIfKeyinfoEmpty(primaryKi, false);
-  if (!primaryKi) {
-    return; // added do_throw=false above + manually exiting here because security.htm can indeed be commonly rendered on setup page before setting acct up
+  setHandlers() {
+    $('.action_change_passphrase').click(this.setHandler(() => Settings.redirectSubPage(this.acctEmail, this.parentTabId, '/chrome/settings/modules/change_passphrase.htm')));
+    $('.action_test_passphrase').click(this.setHandler(() => Settings.redirectSubPage(this.acctEmail, this.parentTabId, '/chrome/settings/modules/test_passphrase.htm')));
+    $('#hide_message_password').change(this.setHandler((el) => this.hideMsgPasswordHandler(el)));
+    $('.password_message_language').change(this.setHandler(() => this.onMsgLanguageUserChange()));
   }
 
-  const storage = await Store.getAcct(acctEmail, ['hide_message_password', 'outgoing_language']);
-  const authInfo = await Store.authInfo(acctEmail);
+  private async renderPPOptionsIfStoredInSession() {
+    let storedInSession = false;
+    const keys = await Store.keysGet(this.acctEmail);
+    for (const key of keys) {
+      if (await Store.passphraseGet(this.acctEmail, key.longid, true)) {
+        storedInSession = true;
+        break;
+      }
+    }
```

Comment:
Actually the current naming is not correct, if I'm seeing correctly, we are testing if pass phrase is stored in permanent local storage, not in session.

This is a good opportunity to move these lines into a separate method, called more precisely `isAnyPassPhraseStoredPermanently`. then below you do `if(await this.isAnyPassPhraseStoredPermanently())`

Reply:
Okay, but if I create a method like `isAnyPassPhraseStoredPermanently(keys)`, will it be okay? This way I won't call `await Store.keysGet(this.acctEmail);` function twice.
Or Store.keysGet(this.acctEmail);` isn't expensive operation?

Reply:
`isAnyPassPhraseStoredPermanently(keys)` is better